nixos/modules/services/networking/ntp/ntpd-rs.nix

```diff
@@ -74,11 +74,64 @@ in
       serviceConfig = {
         User = "";
         Group = "";
-        DynamicUser = true;
         ExecStart = [
           ""
           "${lib.makeBinPath [ cfg.package ]}/ntp-daemon --config=${configFile}"
         ];
+
+        # required to run, without these ntpd-rs reports an error
+        # "Insufficient permissions to interact with the clock"
+        PrivateUsers = false;
+        ProtectClock = false;
+
+        # hardening
+        DynamicUser = true;
+        CapabilityBoundingSet = [
+          "CAP_NET_BIND_SERVICE" # required to bind port 123
+          "CAP_SYS_TIME" # required to set system clock
+        ];
+        AmbientCapabilities = [ "CAP_SYS_TIME" ];
+        RestrictAddressFamilies = [
+          "AF_INET"
+          "AF_INET6"
+          "AF_NETLINK"
```

Comment:
Does ntpd-rs even use Netlink?

Reply:
Without `AF_NETLINK` the service fails to start.

```
systemd[1]: Started Network Time Service (ntpd-rs).
systemd[1]: ntpd-rs.service: Main process exited, code=exited, status=1/FAILURE
systemd[1]: ntpd-rs.service: Failed with result 'exit-code'.
```